pkg/controller/windowsmachineconfig/windows/windows.go

```diff
@@ -256,6 +265,96 @@ func (vm *Windows) ConfigureCNI(configFile string) error {
+func newService(vm *Windows, name, binPath string) *service {
+	return &service{
+		vm:      vm,
+		name:    name,
+		binPath: binPath,
+	}
+}
+
+// start creates and starts the service on the VM
+func (s *service) start() error {
+	_, stderr, err := s.vm.Run("sc.exe create "+s.name+" binPath=\""+s.binPath+"\" start=auto", false)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create service with stderr '%s'", stderr)
+	}
+	_, stderr, err = s.vm.Run("sc.exe start "+s.name, false)
+	if err != nil {
+		return errors.Wrapf(err, "failed to start service with stderr '%s'", stderr)
+	}
+	log.V(1).Info("started service", "name", s.name, "binPath", s.binPath)
+	return nil
+}
+
+// remove removes the service from the VM
+func (s *service) remove() error {
+	_, stderr, err := s.vm.Run("sc.exe delete "+s.name, false)
+	// Exit code 1060 is due to the service not existing. In that case we can return successfully
+	if err != nil && !strings.Contains(err.Error(), "1060 exit code") {
+		return errors.Wrapf(err, "failed to remove service with stderr '%s'", stderr)
+	}
+	return nil
+}
+
+// StartKubeProxyService ensures that the kube-proxy service is running
+func (vm *Windows) StartKubeProxyService(nodeName, hostSubnet string) error {
+	sourceVIP, err := vm.getSourceVIP()
+	if err != nil {
+		return errors.Wrap(err, "error getting source VIP")
+	}
+	// create the kube-proxy log directory
+	if _, stderr, err := vm.Run(mkdirCmd(logDir+"kube-proxy"), false); err != nil {
+		return errors.Wrapf(err, "unable to create %s directory:\n%s", logDir, stderr)
+	}
+	cmd := kubeProxyPath + " --windows-service --v=4 --proxy-mode=kernelspace --feature-gates=WinOverlay=true " +
+		"--hostname-override=" + nodeName + " --kubeconfig=c:\\k\\kubeconfig " +
+		"--cluster-cidr=" + hostSubnet + " --log-dir=" + logDir + "kube-proxy\\ --logtostderr=false " +
+		"--network-name=OVNKubernetesHybridOverlayNetwork --source-vip=" + sourceVIP +
+		" --enable-dsr=false"
+
+	kubeProxy := newService(vm, "kube-proxy", cmd)
+	if err = kubeProxy.remove(); err != nil {
+		return errors.Wrap(err, "error removing kube-proxy service")
+	}
+	if err = kubeProxy.start(); err != nil {
+		return errors.Wrap(err, "error starting kube-proxy service")
+	}
+	return nil
+}
+
+// getSourceVIP returns the source VIP of the VM
+func (vm *Windows) getSourceVIP() (string, error) {
+	cmd := "\"Import-Module -DisableNameChecking " + hnsPSModule + "; " +
+		"$net = (Get-HnsNetwork | where { $_.Name -eq 'OVNKubernetesHybridOverlayNetwork' }); " +
+		"$endpoint = New-HnsEndpoint -NetworkId $net.ID -Name VIPEndpoint; " +
+		"Attach-HNSHostEndpoint -EndpointID $endpoint.ID -CompartmentID 1; " +
+		"(Get-NetIPConfiguration -AllCompartments -All -Detailed | " +
+		"where { $_.NetAdapter.LinkLayerAddress -eq $endpoint.MacAddress }).IPV4Address.IPAddress.Trim()\""
+	stdout, stderr, err := vm.Run(cmd, true)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get source VIP with stderr '%s'", stderr)
```

Comment:
'%s' is this in single quotes for some reason?

Reply:
Just formatting to show that it is all one string and not part of the generic error message

Reply:
How about using `:` instead of the quotes?